datafusion/functions-array/src/string.rs

```diff
@@ -281,6 +281,21 @@ pub(super) fn array_to_string_inner(args: &[ArrayRef]) -> Result<ArrayRef> {
 
                 Ok(arg)
             }
+            Dictionary(..) => {
+                let any_dict_array = arr.as_any_dictionary_opt().ok_or_else( || {
+                    DataFusionError::Internal(
+                        format!("could not cast {} to AnyDictionaryArray", arr.data_type())
+                    )
+                })?;
+                compute_array_to_string(
```

Comment:
I think this code may just return the dictionary converted to string, 

https://docs.rs/arrow/latest/arrow/array/struct.DictionaryArray.html


You might have to expand out the values again (perhaps cast the result to `DataType::Utf8` 🤔

Reply:
Checking

Reply:
I'll continue the conversation in the comments below